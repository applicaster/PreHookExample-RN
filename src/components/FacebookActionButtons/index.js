import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  View,
 } from 'react-native';
import { styles } from './style';
import EventDetailCount from '../EventDetailCount';

class FacebookActionButtons extends Component {
  render() {
    const { commentsCount, likesCount, eventId } = this.props;
    return (
      <View style={[styles.facebookActionButtons]}>
        <EventDetailCount key={'likes'} eventId={eventId} count={likesCount} label={'Likes'} />
        <EventDetailCount key={'comments'} eventId={eventId} count={commentsCount} label={'Comments'} />
      </View>
    );
  }
}

FacebookActionButtons.propTypes = {
  eventId: PropTypes.string,
  commentsCount: PropTypes.number,
  likesCount: PropTypes.number,
};

export default FacebookActionButtons;
