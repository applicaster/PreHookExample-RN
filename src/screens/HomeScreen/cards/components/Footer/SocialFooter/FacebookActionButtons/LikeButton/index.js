import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setActiveEventId } from '../../../../../../../../actions';

import LikeButton from './LikeButton';

const mapDispatchToProps = dispatch => bindActionCreators({
  setActiveEventId,
}, dispatch);

export default connect(undefined, mapDispatchToProps)(LikeButton);
