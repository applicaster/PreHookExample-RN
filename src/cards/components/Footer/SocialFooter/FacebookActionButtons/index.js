import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  View,
 } from 'react-native';
import { styles } from './style';
import LikeButton from './LikeButton';
import CommentButton from './CommentButton';

class FacebookActionButtons extends Component {
  render() {
    const { commentsCount, likesCount, eventId } = this.props;
    return (
      <View style={[styles.facebookActionButtons]}>
        <LikeButton key={'likes'} eventId={eventId} likesCount={likesCount} />
        <CommentButton key={'comments'} eventId={eventId} commentsCount={commentsCount} />
      </View>
    );
  }
}

FacebookActionButtons.propTypes = {
  eventId: PropTypes.string.isRequired,
  commentsCount: PropTypes.number.isRequired,
  likesCount: PropTypes.number.isRequired,
};

export default FacebookActionButtons;
