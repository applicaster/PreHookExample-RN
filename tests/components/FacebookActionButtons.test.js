import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import FacebookActionButtons from '../../src/components/FacebookActionButtons';

jest.mock('../../src/components/EventDetailCount', () => 'EventDetailCount');
jest.mock('@applicaster/feed-rn-utils', () => 'FeedRNUtils');

describe('FacebookActionButtons', () => {
  test('component renders correctly', () => {
    const html = renderer.create(
      <FacebookActionButtons />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
