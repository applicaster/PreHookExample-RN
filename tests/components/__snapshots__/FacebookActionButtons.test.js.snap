// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacebookActionButtons component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
>
  <EventDetailCount
    count={undefined}
    eventId={undefined}
    label="Likes"
  />
  <EventDetailCount
    count={undefined}
    eventId={undefined}
    label="Comments"
  />
</View>
`;
